Sources/SourceKitLSP/SourceKitLSPServer.swift

```diff
@@ -612,7 +612,7 @@ extension SourceKitLSPServer: MessageHandler {
     // Keep track of the ID -> Task management with low priority. Once we cancel
     // a request, the cancellation task runs with a high priority and depends on
     // this task, which will elevate this task's priority.
-    cancellationMessageHandlingQueue.async(priority: .background) {
+    cancellationMessageHandlingQueue.async(priority: .high) {
```

Comment:
`cancellationMessageHandlingQueue` is an `AsyncQueue`, so shouldn't that enforce that this runs before handling a subsequent cancellation?

Reply:
Now that you mention it: I did think about this when I wrote the code originally. And also looking at the logs again, it seems like the issue is that we just returned results. I’ll open a new PR that fixes that issue instead.